src/apis/objkt/index.ts

```diff
@@ -115,18 +121,114 @@ export const fetchCollectiblesByCollection$ = (
   );
 };
 
-export const fetchCollectibleInfo$ = (address: string, tokenId: string): Observable<CollectibleInfo> => {
-  const request = buildGetCollectibleByAddressAndIdQuery(address, tokenId);
+export const fetchAllCollectiblesDetails$ = (collectiblesSlugs: string[]): Observable<CollectibleDetailsResponse[]> => {
+  const request = buildGetAllUserCollectiblesQuery(collectiblesSlugs);
+
+  return apolloObjktClient.query<UserAdultCollectiblesQueryResponse>(request).pipe(
+    map(result => {
+      return result.token.map(
+        ({
+          fa_contract,
+          token_id,
+          name,
+          description,
+          thumbnail_uri,
+          artifact_uri,
+          creators,
+          fa,
+          timestamp,
+          attributes,
+          tags,
+          metadata,
+          royalties,
+          supply,
+          listings_active,
+          mime,
+          galleries
+        }) => ({
+          fa_contract,
+          token_id,
+          name,
+          description,
+          thumbnail_uri,
+          artifact_uri,
+          creators,
+          fa: {
+            name: fa.name,
+            logo: fa.logo,
+            items: fa.items
+          },
+          metadata,
+          attributes,
+          tags,
+          timestamp,
+          royalties,
+          supply,
+          galleries,
+          listings_active,
+          mime
+        })
+      );
+    })
+  );
+};
+
+export const fetchFA2AttributeCount$ = (ids: number[]): Observable<AttributeInfoResponse[]> => {
+  const request = buildGetFA2AttributeCountQuery(ids);
+
+  return apolloObjktClient
+    .query<FA2AttributeCountQueryResponse>(request)
+    .pipe(map(result => getUniqueAndMaxValueAttribute(result.fa2_attribute_count)));
+};
+
+export const fetchGalleryAttributeCount$ = (ids: number[]): Observable<AttributeInfoResponse[]> => {
+  const request = buildGetGalleryAttributeCountQuery(ids);
+
+  return apolloObjktClient
+    .query<GalleryAttributeCountQueryResponse>(request)
+    .pipe(map(result => getUniqueAndMaxValueAttribute(result.gallery_attribute_count)));
+};
 
-  return apolloObjktClient.query<CollectibleInfoQueryResponse>(request).pipe(
+export const fetchCollectibleFloorPrice$ = (address: string, id: string): Observable<ListingsActive[]> => {
+  const request = buildGetCollectibleFloorPriceQuery(address, id);
+
+  return apolloObjktClient.query<CollectibleFloorPriceQueryResponse>(request).pipe(
+    map(result => {
+      const { listings_active } = result.token[0];
+
+      return isNonEmptyArray(listings_active)
+        ? [
+            {
+              bigmapKey: listings_active[0].bigmap_key ?? 0,
+              currency: listings_active[0].currency,
+              currencyId: listings_active[0].currency_id,
+              marketplaceContract: listings_active[0].marketplace_contract,
+              price: listings_active[0].price
+            }
```

Comment:
Magic zero

Reply:
Fix here and in all another places please

Reply:
it means the first item. I'm convinced we don't need to create separate variable for this case